### ISSUE TITLE
`RouteAnnouncer` does not announce new routes 

### ISSUE THREAD

**Issue by sylvhama** [gatsbyjs/gatsby#35187]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

We should be able to tell to the [RouteAnnouncer](https://www.gatsbyjs.com/blog/2020-02-10-accessible-client-side-routing-improvements/#new-improvements) what string to use when our website or web app has a static h1 that does not change after navigating to a new page. 

I've made a [draft PR](https://github.com/gatsbyjs/gatsby/pull/35186/files) to illustrate a potential solution. The idea would be to check for a specific id first. If it's not there then we could default to the current logic. This is not the most elegant way, but since each website / web app has its own HTML structure, this would give the freedom to choose the element that contains the new page title without having to import a custom hook or another component. E.g. on my personal website I have a visually hidden h2: https://github.com/sylvhama/shamann-gatsby/blob/master/src/components/Section.js#L14.

Let me know if that direction would be okay, I would be more than happy to polish my draft PR.

Another more complex idea would be to create a `RouteAnnouncerContext` that would have a state value storing the string rendered within the live region. Then in each page we could call a custom hook (`useRouteAnnouncer`) triggering a side effect to change the state.

### Reproduction Link

https://gatsbystarterdefaultsource.gatsbyjs.io/

### Steps to Reproduce

1. Visit https://gatsbystarterdefaultsource.gatsbyjs.io/
2. Start a screen reader (e.g. VoiceOver on Mac OS) or use https://assistivlabs.com/
3. Navigate to different pages
4. The screen reader will announce "Navigated to Gatsby Default Starter" only once then nothing;

### Expected Result

We should hear the content of the second `h1` e.g. "Navigated to Gatsby supports TypeScript by default!". 

### Actual Result

It will announce "Navigated to Gatsby Default Starter" after the first navigation then it won't announce anything else.

### Environment

```shell
NA
```
